src/App.tsx

```diff
@@ -8,37 +8,168 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import {
+  getUsers,
+  getUserPosts,
+  getPostComments,
+  createComment,
+  deleteComment,
+} from './api/Users';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { Comment } from './types/Comment';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [userPosts, setUserPosts] = useState<Post[] | null>(null);
+  const [postComments, setPostComments] = useState<Comment[] | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [isCommentsError, setCommentsError] = useState(false);
+  const [activePost, setActivePost] = useState<Post | null>(null);
+  const [isNewCommentForm, setIsNewCommentForm] = useState(false);
+  const [isCommentLoading, setIsCommentLoading] = useState(false);
+
+  function clearSidebar() {
+    setIsError(false);
+    setCommentsError(false);
+    setIsNewCommentForm(false);
+  }
+
+  const loadUsers = async () => {
+    try {
+      clearSidebar();
+      const usersFromServer = await getUsers();
+
+      setUsers(usersFromServer);
+    } catch (error) {
+      setIsError(true);
+    }
+  };
+
+  const loadUserPosts = async (userId: number) => {
+    try {
+      setIsLoading(true);
+      setActivePost(null);
+      clearSidebar();
+      const posts = await getUserPosts(userId);
+
+      setUserPosts(posts);
+    } catch (error) {
+      setIsError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const loadComments = async (postId: number) => {
+    try {
+      clearSidebar();
+      const comments = await getPostComments(postId);
+
+      setPostComments(comments);
+    } catch (error) {
+      setCommentsError(true);
+    }
+  };
+
+  const postComment = async (data: Omit<Comment, 'id'>) => {
+    try {
+      setIsCommentLoading(true);
+      const newComment = await createComment(data);
+
+      setPostComments(comments => {
+        return (
+          comments !== null ? ([
+            ...comments,
+            newComment,
+          ]) : (
+            [newComment]
+          ));
+      });
```

Comment:
```suggestion
      setPostComments(comments => (comments !== null 
        ? [
            ...comments,
            newComment,
          ] 
        : [newComment];
      );
```
Add brackets if needed